Base/QTGUI/qSlicerWebDownloadWidget.cxx

```diff
@@ -0,0 +1,141 @@
+/*==============================================================================
+
+  Program: 3D Slicer
+
+  Copyright (c) Kitware Inc.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Steve Pieper, Isomics Inc.
+  and was partially funded by NSF grant DBI 1759883
+
+==============================================================================*/
+
+// Qt includes
+#include <QDebug>
+#include <QDir>
+#include <QFileInfo>
+#include <QFileDialog>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QMessageBox>
+#include <QProgressBar>
+#include <QPushButton>
+#include <QVBoxLayout>
+#include <QWebEngineDownloadItem>
+
+// Slicer includes
+#include "qSlicerCoreApplication.h"
+#include "qSlicerDataDialog.h"
+#include "qSlicerWebDownloadWidget.h"
+
+// --------------------------------------------------------------------------
+qSlicerWebDownloadWidget::qSlicerWebDownloadWidget(QWidget *parent)
+  : QDialog(parent)
+{
+}
+
+// --------------------------------------------------------------------------
+void qSlicerWebDownloadWidget::handleDownload(QWebEngineDownloadItem *download)
+{
+
+  // need to use a modal dialog here because 'download' will be deleted
+  // if we don't return 'accept from this slot
+  QMessageBox *messageBox = new QMessageBox(this);
```

Comment:
Below is the specific note in the code about this. Also background discussion can be found in #951.
https://github.com/Slicer/Slicer/blob/5d70815798b8f324b2a14ad6e9cfa1634edf9ba9/Base/Python/slicer/util.py#L1130-L1136

Reply:
It seems to me the WA_DeleteOnClose attribute idea is the right one here, since this dialog shows progress/cancel information that exists through the event loop, while [deleteLater](http://doc.qt.io/qt-5/qobject.html#deleteLater) says it deletes once the event loop is entered.

Reply:
Actually WA_DeleteOnClose wasn't right because this confirmation dialog is used later.  

But I think it's actually deleted because I set 'this' to be it's parent, so that should close it up when the download dialog is deleted.

In the util.py case the dialog has a default parent of None (or maybe people set the parent to be mainWindow) so it would stay around.

Reply:
Yes, if the parent gets deleted (make sure it does, if it's a dialog then it might not) then it should work well without delete-on-close.

Reply:
Good point - the parent is the download progress widget, which is a child of the webwidget so it would have been deleted eventually, but I guess it's best to delete it when the download is complete to avoid the lingering window issue.  I've updated to that and it seems to work.